### PR TITLE
Implement easy access for the current_user's vacols uniq ID and more …

### PR DESCRIPTION
**Pull request by aroltsch**
…performance improvements

1. Better error handling
2. Performance improvements
3. Easy access to the current_user's SLOGID

### CODE REVIEW

File: app/controllers/issues_controller.rb [hunk 1]
```diff
@@ -42,17 +39,23 @@ def destroy
...
-                                    :level_2,
-                                    :level_3)
+    params.require("issues")
+      .permit(:note,
+              :program,
+              :issue,
+              :level_1,
+              :level_2,
+              :level_3)
+      .merge!(vacols_user_id: current_user.vacols_uniq_id)
```
Comment (aroltsch) [inline app/controllers/issues_controller.rb]: Query for slogid only once.

File: app/controllers/issues_controller.rb [hunk 2]
```diff
@@ -42,17 +39,23 @@ def destroy
 
   private
 
+  def validate_access_to_task
+    current_user.access_to_task?(appeal.vacols_id)
```
Comment (aroltsch) [inline app/controllers/issues_controller.rb]: Move this out to the issues controller in order to improve performance when creating a draft decision/omo request.

File: app/repositories/issue_repository.rb [hunk 3]
```diff
@@ -1,15 +1,10 @@
 class IssueRepository
-  class IssueError < StandardError; end
-
   # :nocov:
-  def self.create_vacols_issue!(css_id:, issue_attrs:)
+  def self.create_vacols_issue!(issue_attrs:)
     MetricsService.record("VACOLS: create_vacols_issue! for #{issue_attrs[:vacols_id]}",
                           service: :vacols,
                           name: "IssueRepository.create_vacols_issue!") do
-      validate_access!(css_id, issue_attrs[:vacols_id])
```
Comment (aroltsch) [inline app/repositories/issue_repository.rb]: Moved this to the controller.

File: lib/caseflow/error.rb [hunk 4]
```diff
@@ -32,4 +32,10 @@ def self.from_vbms_error(error)
   end
 
   class DuplicateEp < EstablishClaimFailedInVBMS; end
+
+  class VacolsRepositoryError < StandardError; end
```
Comment (aroltsch) [inline lib/caseflow/error.rb]: All repository errors will inherit from VacolsRepositoryError so we can just catch one type and not worry about specifics.

Comment (amprokop) [reply lib/caseflow/error.rb]: 👍

File: app/controllers/issues_controller.rb (hunk 1 above)
Comment (amprokop) [reply app/controllers/issues_controller.rb]: Nice!

**Pull request merged**